code/modules/mob/living/carbon/human/ai/firearm_appraisal.dm

```diff
@@ -190,9 +203,17 @@ GLOBAL_LIST_INIT_TYPED(firearm_appraisals, /datum/firearm_appraisal, build_firea
 		/obj/item/weapon/gun/launcher/rocket/anti_tank/disposable,
 	)
 	disposable = TRUE
+	primary_weight = 15
 
 /datum/firearm_appraisal/rpg/multi_use
 	gun_types = list(
 		/obj/item/weapon/gun/launcher/rocket,
 	)
 	disposable = FALSE
+
+/datum/firearm_appraisal/pistol
+	maximum_range = 6
+	gun_types = list(
+		/obj/item/weapon/gun/pistol,
```

Comment:
Pistols are a thing that can be often replaced by a revolver in equipment presets, so considering it will have similar appraisal to this one, i'd recommend adding `/obj/item/weapon/gun/revolver` to the list
```suggestion
		/obj/item/weapon/gun/pistol,
		/obj/item/weapon/gun/revolver,
```